docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,692 @@
+
+#### Example: Set Custom Request Headers
+
+**Setting Default Custom Request Headers in HttpClient**
+
+```csharp
+var client = new HttpClient();
+client.DefaultRequestHeaders.Add("Custom-Header", "value");
+```
+
+**Setting Custom Request Headers in HttpRequestMessage**
+
+```csharp
+var request = new HttpRequestMessage(HttpMethod.Get, uri);
+request.Headers.Add("Custom-Header", "value");
+```
+
+#### Example: Set Common Request Headers
+
+When working with `HttpRequestMessage` in .NET, setting common request headers is essential for providing additional information to the server about the request being made. These headers can include authentication tokens, content types and more. Properly configuring these headers ensures that your HTTP requests are processed correctly by the server.
+For a comprehensive list of common properties available in HttpRequestHeaders, you can refer to the [official documentation](https://learn.microsoft.com/dotnet/api/system.net.http.headers.httprequestheaders#properties).
+
+To set common request headers in `HttpRequestMessage`, you can use the `Headers` property of the `HttpRequestMessage` object. This property provides access to the `HttpRequestHeaders` collection, where you can add or modify headers as needed.
+
+**Setting Common Default Request Headers in HttpClient**
+
+```csharp
+using System.Net.Http.Headers;
+
+var client = new HttpClient();
+client.DefaultRequestHeaders.Authorization =  new AuthenticationHeaderValue("Bearer", "token");
+```
+
+**Setting Common Request Headers in HttpRequestMessage**
+
+```csharp
+using System.Net.Http.Headers;
+
+var request = new HttpRequestMessage(HttpMethod.Get, uri);
+request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", "token");
+```
+
+### Example: Set Content Headers
+
+Content headers are used to provide additional information about the body of an HTTP request or response. When working with `HttpClient` in .NET, you can set content headers to specify the media type, encoding, and other metadata related to the content being sent or received. Properly configuring content headers ensures that the server and client can correctly interpret and process the content.
+
+```csharp
+var client = new HttpClient();
+var request = new HttpRequestMessage(HttpMethod.Post, uri);
+
+// Create the content and set the content headers
+var jsonData = "{\"key\":\"value\"}";
+var content = new StringContent(jsonData, Encoding.UTF8, "application/json");
+content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+content.Headers.ContentEncoding.Add("utf-8");
+content.Headers.ContentLength = jsonData.Length;
```

Comment:
```suggestion
```

I think we should avoid giving an example for this one.
It's unlikely that the user actually needs it (in this example, `StringContent` already calculates it for you), and it's easy to get wrong.

E.g. in this example, if you change `jsonData` to contain non-ASCII text, you'll break the content since the length will be wrong (number of UTF-16 chars instead of UTF-8 bytes).

Reply:
I think we can still keep this example here with extra warning like you said, but we should also add a note that they will not need this and Content itself will calculate `ContentLength` automatically based on the content data.